homeassistant/components/zwave_js/fan.py

```diff
@@ -111,9 +120,90 @@ def percentage_step(self) -> float:
     @property
     def speed_count(self) -> int:
         """Return the number of speeds the fan supports."""
-        return int_states_in_range(SPEED_RANGE)
+        return int_states_in_range(DEFAULT_SPEED_RANGE)
 
     @property
     def supported_features(self) -> int:
         """Flag supported features."""
         return SUPPORTED_FEATURES
+
+
+class ConfiguredSpeedRangeZwaveFan(ZwaveFan):
+    """A Zwave fan with a configured speed range (e.g., 1-24 is low)."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize the fan."""
+        super().__init__(config_entry, client, info)
+        self.data_template = cast(
+            FanSpeedDataTemplate, self.info.platform_data_template
+        )
+
+    async def async_set_percentage(self, percentage: int | None) -> None:
```

Comment:
```suggestion
    async def async_set_percentage(self, percentage: int) -> None:
```

Reply:
Done (in conjunction with the async_turn_on changes)